src/UI/toolbar/BatteryIndicator.qml

```diff
@@ -278,6 +322,117 @@ Item {
                         }
                     }
                 }
+                
+            }
+
+            SettingsGroupLayout {
+                heading:        qsTr("Battery State Display")
+                Layout.fillWidth: true
+                spacing: ScreenTools.defaultFontPixelWidth * 0.5
```

Comment:
The visibility of this group should be based on `batterySettings.visible`. This allows custom builds to hide this section if they want to.

Also each individual settings has a `visible` property.  If not visible then the editing field would normally be completely hidden. But in this caseI would show just a label with the percentage instead of a configurable text field.

Both of these combined give a good level of custom build override capability.

Reply:
I’ve implemented the visibility controls as you suggested, using `visible: batterySettings.visible` for the entire group and `bool thresholdVisible: batterySettings.thresholdEditable` for individual settings.

In the `BatteryIndicatorSettings` class, I’ve added QSettings to manage visibility and editability, defaulting to `true` if no value is set:

```cpp
bool BatteryIndicatorSettings::visible() const {
    QSettings settings;
    return settings.value("ShowBatterySettings", true).toBool();
}

bool BatteryIndicatorSettings::thresholdEditable() const {
    QSettings settings;
    return settings.value("ThresholdEditable", true).toBool();
}
```
However, I’m unsure how to properly configure these two settings to effectively allow custom builds to hide this section if they choose to. Could you provide how to set this up correctly? Your expertise would be greatly appreciated!

Reply:
In the current setup, if the `thresholdVisible` property is **false**, the `FactTextField` for `threshold1` and `threshold2` will be hidden, and a `QGCLabel` will display the current percentages. This ensures that users can still see the battery thresholds without the editing fields when they're not meant to be configurable.

![Screenshot from 2024-09-23 21-15-09](https://github.com/user-attachments/assets/2a4d9c40-85a5-4021-bcc8-f3380ac79ca1)

If the `thresholdVisible` property is **true**, the `FactTextField` for both `threshold1` and `threshold2` will be visible, allowing users to edit the thresholds directly. 

![Screenshot from 2024-09-23 21-23-33](https://github.com/user-attachments/assets/64b4a5dd-d06e-4e29-ac98-7ec43d9aaf7b)

Reply:
> However, I’m unsure how to properly configure these two settings to effectively allow custom builds to hide this section

there is no extra work needed. SettingsGroup and SettingsFact base classes already support this automatically.